seed/generateData.js

```diff
@@ -0,0 +1,52 @@
+const fs = require('fs');
+const path = require('path');
+const csvWriter = require('csv-write-stream');
+const writer = csvWriter();
+const Page = require('./Page.js');
+const filePath = path.join(__dirname, '/data');
+
+const timer = (record, start) => {
+  if (record % 100000 === 0) {
+    const now = Date.now();
+    const minutesElapsed = (((now - start) / 1000) / 60);
+    console.log(`created record #${record} - minutes elapsed: ${minutesElapsed}`);
+  }
+}
+
+const start = Date.now();
+
+writer.pipe(fs.createWriteStream(filePath + '/pages.csv'));
+
+module.exports.generateCSV = async () => {
+  for (let i = 1; i < 1e7 ; i++) {
+    writer.write((new Page(i)));
```

Comment:
based on my understanding, setTimeout of 0ms and setImmediate behave the same exact way. in both cases, the function will be placed at the end of the current event loop.

i think im going to try to avoid going down the promise route. i did some reading on node fs module and i want to reconfigure to use drain properly.